csrc/options.h

```diff
@@ -114,6 +114,27 @@ enum class DisableOption {
   EndOfOption //! Placeholder for counting the number of elements
 };
 
+//! Options to set for Fusion Profiling.  Whenever the profiler
+//! is enabled, its output can be queried from the FusionProfile object.
+//! All options enable the profiler.
+//!
+//! These can be set through the `NVFUSER_PROF` environment variable
+//!
+enum class ProfilerOption {
+  Enable, //! Enables the profiler.
+  EnableNocupti, //! Enables the profiler, but disables CUPTI specific
+                 //! profiling inorder to measure true host time without
+                 //! overhead.
+  Print, //! Enables the profiler and prints the output to the console.
+  PrintNocupti, //! Enables the profiler, disables CUPTI specific
+                //! profiling inorder to measure true host time without
+                //! overhead, and prints the output to the console.
+  PrintVerbose, //! Enables the profiler and prints a complete set of columns
+                //! to the console.  WARNING: The output is will wrap on small
```

Comment:
I think in theory we are missing PrintNoCuptiVerbose? It's unfortunate we have to list all combinations. Is it possible to decouple enablement and printing so we have fewer knobs?

Reply:
`print.nocupti.verbose` does not make any sense as `.verbose` and `.nocupti` are distinct.  `.nocupti` prints a very limited subset of statistic only at the Fusion level that does not include any kernel specific timing.  `.verbose` is used to print out a larger set of of columns of statistics.  The goal with `.verbose` is to remove stats that don't change that and help to fit the set of columns on smaller screens.  You can see examples of how the 3 differ in 1st comment/description of the PR.

You can see the `.verbose` knob sets a variable in the `FusionProfile`: https://github.com/NVIDIA/Fuser/blob/baaacdf36e87416ca8692b0e9fb72192e08c6e89/csrc/fusion_profiler.h#L133

This is for use with the `operator<<()` for `FusionProfile`. 

https://github.com/NVIDIA/Fuser/blob/baaacdf36e87416ca8692b0e9fb72192e08c6e89/csrc/fusion_profiler.cpp#L529

Reply:
I understood that -- that's why I said "in theory" we are missing PrintNoCuptiVerbose. My point was `Enable*` and `Print*` are orthogonal and therefore map more clearly to two separate flags. If we do that, `PrintNoCupti` doesn't even need to exist because it can be represented as `EnableNoCupti`+`Print`. 

Anyhow, the practical benefit of removing `PrintNoCupti` may be too small to justify further changes to this PR.